### PR TITLE
Device settings

### PR DESCRIPTION
**Pull request by StimkaT**

### CODE REVIEW

File: projects/blink/src/lib/+state/blink-commands-list/blink-commands-list.reducer.ts [hunk 1]
```diff
@@ -17,9 +16,9 @@ export interface BlinkCommandsListPartialState {
 }
 
 export const initialState: BlinkCommandsListState = {
-    commands: [ {num: 122, name: 'nam1', event: 'ev1'}, 
-    {num: 222, name: 'nam2', event: 'ev2'},
-    {num: 222123123, name: 'nam2', event: 'ev2'}]
+    commands: [ {name: 'ON Indicator', event: '{"event":"LED","data":{"command":"ON"}}'}, 
+    {name: 'OFF Indicator', event: '{"event":"LED","data":{"command":"OFF"}}'},
+    {name: 'BLINK Indicator', event: '{"event":"LED","data":{"command":"BLINL"}}'}]
```
Comment (StimkaT) [inline projects/blink/src/lib/+state/blink-commands-list/blink-commands-list.reducer.ts:21]: BLINL to BLINK

File: projects/app/src/lib/+state/commands-list/commands-list.reducer.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@ import { createReducer } from '@ngrx/store';
 export const COMMANDS_LIST_FEATURE_KEY = 'general commands';
```
Comment (StimkaT) [inline projects/app/src/lib/+state/commands-list/commands-list.reducer.ts:3]: general commands to comandsList

File: projects/app/src/lib/+state/commands-list/commands-list.reducer.ts [hunk 3]
```diff
@@ -17,8 +16,8 @@ export interface CommandsListPartialState {
 }
 
 export const initialState: CommandsListState = {
-    commands: [ {num: 1, name: 'nam1', event: 'ev1'},
```
Comment (StimkaT) [inline projects/app/src/lib/+state/commands-list/commands-list.reducer.ts:20]: commands to comandsList

File: projects/app/src/lib/+state/commands-list/commands-list.reducer.ts [hunk 4]
```diff
@@ -17,8 +16,8 @@ export interface CommandsListPartialState {
 }
 
 export const initialState: CommandsListState = {
-    commands: [ {num: 1, name: 'nam1', event: 'ev1'}, 
-    {num: 2, name: 'nam2', event: 'ev2'}]
+    commands: [ {name: 'Connect to device', event: '{"event":"CONNECT_USB_DEVICE","data":{"name":"COM1","timestamp":1706869676852}}'},
```
Comment (StimkaT) [inline projects/app/src/lib/+state/commands-list/commands-list.reducer.ts:19]: commands to comandsList

File: projects/blink/src/lib/+state/blink-commands-list/blink-commands-list.reducer.ts [hunk 5]
```diff
@@ -3,7 +3,6 @@ import { createReducer } from '@ngrx/store';
 export const BLINK_COMMANDS_LIST_FEATURE_KEY = 'blink commands';
```
Comment (StimkaT) [inline projects/blink/src/lib/+state/blink-commands-list/blink-commands-list.reducer.ts:3]: blink commands to blink\commandsList

File: projects/blink/src/lib/+state/blink-mode/blink-mode.reducer.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { createReducer, on } from '@ngrx/store';
+import * as BlinkModeActions from './blink-mode.actions';
+
+export const BLINK_MODE_FEATURE_KEY = 'blinkMode';
```
Comment (StimkaT) [inline projects/blink/src/lib/+state/blink-mode/blink-mode.reducer.ts:4]: blinkMode to blink\mode

File: projects/blink/src/lib/containers/blink-container/blink-container.component.html [hunk 7]
```diff
@@ -1,6 +1,5 @@
-<page-layout>
-  <blink-nav-panel-container  toplBar></blink-nav-panel-container>
-  <div body>
-    <light  (emitter)="events($event)"></light>
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-container/blink-container.component.html:4]: 2 space

File: projects/blink/src/lib/containers/blink-container/blink-container.component.ts [hunk 8]
```diff
@@ -1,52 +1,41 @@
...
 export class BlinkContainerComponent implements OnInit {
+
+  mode$ = this.store$.select(getMode);
+
   constructor(
     private readonly store: Store,
-    private route: ActivatedRoute
-  ) {
+    private route: ActivatedRoute,
+    private store$: Store<BlinkModeState>){
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-container/blink-container.component.ts:34]: private store$: Store<BlinkModeState> to     private store$: Store

File: projects/blink/src/lib/containers/blink-nav-panel-container/blink-nav-panel-container.component.ts [hunk 9]
```diff
@@ -5,6 +5,7 @@ import {Store} from '@ngrx/store';
 import {Router} from '@angular/router';
 // import { setData } from '../../../../../app/src/lib/+state/about/about.actions';
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-nav-panel-container/blink-nav-panel-container.component.ts:6]: remove comment

File: projects/blink/src/lib/containers/blink-nav-panel-container/blink-nav-panel-container.component.ts [hunk 10]
```diff
@@ -5,6 +5,7 @@ import {Store} from '@ngrx/store';
...
 import { sendMessage } from '../../../../../app/src/lib/+state/messages/messages.actions';
+import { setMode } from '../../+state/blink-mode/blink-mode.actions';
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-nav-panel-container/blink-nav-panel-container.component.ts:9]: add empty line

File: projects/blink/src/lib/containers/blink-nav-panel-container/blink-nav-panel-container.component.ts [hunk 11]
```diff
@@ -34,6 +35,10 @@ export class BlinkNavPanelContainerComponent {
       this.router.navigate(['control/blink/:deviceName/logDevice']);
     } else  if ($event.event === 'NavDeviceComponent:BUTTON_CLICKED' && $event.data === 'commandsList') {
       this.router.navigate(['control/blink/:deviceName/commandsList']);
+    } else  if ($event.event === 'NavDeviceComponent:BUTTON_CLICKED' && $event.data === 'two') {
+      this.store.dispatch(setMode({mode: 'two'}));
+    } else  if ($event.event === 'NavDeviceComponent:BUTTON_CLICKED' && $event.data === 'three') {
+      this.store.dispatch(setMode({mode: 'three'}));
     } 
     // else  if ($event.event === 'NavDeviceComponent:BUTTON_CLICKED' && $event.data === 'aboutDevice') {
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-nav-panel-container/blink-nav-panel-container.component.ts:43]: remove comment

File: projects/blink/src/lib/containers/blink-three-container/blink-three-container.component.html [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<page-layout>
+  <blink-nav-panel-container  toplBar></blink-nav-panel-container>
+  <div body>
+    <light-three class="qq" (emitter)="events($event)"></light-three>
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-three-container/blink-three-container.component.html:4]: rename class

File: projects/blink/src/lib/containers/blink-three-container/blink-three-container.component.ts [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+@Component({
+  selector: 'blink-three-container',
+  standalone: true,
+  imports: [
+    PageLayoutComponent,
+    BlinkNavPanelContainerComponent,
+    LightThreeComponent,
+    RouterOutlet,
+    AsyncPipe,
+    BlinkAboutContainerComponent
```
Comment (StimkaT) [inline projects/blink/src/lib/containers/blink-three-container/blink-three-container.component.ts:23]: add comma

File: projects/ui/src/lib/components/commands-list/commands-list.component.html [hunk 14]
```diff
@@ -2,25 +2,25 @@
...
-    <th mat-header-cell *matHeaderCellDef> Name event </th>
-    <td mat-cell *matCellDef="let element"> {{element.name}} </td>
+        <th mat-header-cell *matHeaderCellDef> Name event </th>
+        <td mat-cell *matCellDef="let element"> {{element.name}} </td>
     </ng-container>
 
-    <ng-container matColumnDef="event">
-    <th mat-header-cell *matHeaderCellDef> Description event </th>
-    <td mat-cell *matCellDef="let element"> {{element.event}} </td>
+    <ng-container  matColumnDef="event">
```
Comment (StimkaT) [inline projects/ui/src/lib/components/commands-list/commands-list.component.html:14]: 2 space

File: projects/ui/src/lib/components/light-three-button/light.component.html [hunk 15]
```diff
@@ -1,25 +1,25 @@
-<mat-grid-list cols="3" rowHeight="2:1">
+<mat-grid-list  cols="3" rowHeight="2:1">
```
Comment (StimkaT) [inline projects/ui/src/lib/components/light-three-button/light.component.html:1]: 2 space

File: projects/ui/src/lib/components/light-three-button/light.component.html [hunk 16]
```diff
@@ -1,25 +1,25 @@
...
     <mat-grid-tile>
         <img src="./assets/images/blink/off.png">
     </mat-grid-tile>
     <mat-grid-tile>
-        <mat-icon>auto_awesome</mat-icon>
+        <img src="./assets/images/blink/blink.png">
     </mat-grid-tile>
     <mat-grid-tile>
-        <button (click)="buttonClick('ON')" >
+        <button mat-raised-button color="primary" (click)="buttonClick('ON')" >
```
Comment (StimkaT) [inline projects/ui/src/lib/components/light-three-button/light.component.html:12]: remove space

File: projects/ui/src/lib/components/light-three-button/light.component.scss [hunk 17]
```diff
@@ -0,0 +1,9 @@
+button {
+    font-size: 15px;
+    width: 200px;
+}
+
```
Comment (StimkaT) [inline projects/ui/src/lib/components/light-three-button/light.component.scss:5]: empty line

File: projects/ui/src/lib/components/light-two-button/light.component.html [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    Light ON
+</button>
+<button mat-raised-button color="primary" (click)="buttonClick('OFF')">
+    Light OFF
+</button> 
+<button mat-raised-button color="primary" (click)="buttonClick('BLINK')">
+    Blink
+</button>
+} @else if (led==='OFF') {
+    <button mat-raised-button color="primary" (click)="buttonClick('ON')" >
```
Comment (StimkaT) [inline projects/ui/src/lib/components/light-two-button/light.component.html:21]: remove space

File: projects/app/src/lib/containers/main-container/main-container.component.ts [hunk 19]
```diff
@@ -16,8 +19,11 @@ import {RouterOutlet} from '@angular/router';
 export class MainContainerComponent {
   getUsbList$ = this.store.pipe(select(getUsbList));
 
+  mode$ = this.store$.select(getMode);
+
   constructor(
     private readonly store: Store,
+    private store$: Store<BlinkModeState>
```
Comment (sislex) [inline projects/app/src/lib/containers/main-container/main-container.component.ts:26]: add coma

File: projects/blink/src/lib/+state/blink-commands-list/blink-commands-list.reducer.ts (hunk 5 above)
Comment (sislex) [reply projects/blink/src/lib/+state/blink-commands-list/blink-commands-list.reducer.ts:3]: blink/commandsList

File: projects/blink/src/lib/+state/blink-mode/blink-mode.reducer.ts (hunk 6 above)
Comment (sislex) [reply projects/blink/src/lib/+state/blink-mode/blink-mode.reducer.ts:4]: blink/mode

File: projects/blink/src/lib/containers/blink-container/blink-container.component.html [hunk 20]
```diff
@@ -1,6 +1,5 @@
-<page-layout>
-  <blink-nav-panel-container  toplBar></blink-nav-panel-container>
```
Comment (sislex) [inline projects/blink/src/lib/containers/blink-container/blink-container.component.html:2]: 2 space

File: projects/blink/src/lib/containers/blink-container/blink-container.component.ts [hunk 21]
```diff
@@ -1,52 +1,41 @@
...
   styles: ``,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class BlinkContainerComponent implements OnInit {
+
+  mode$ = this.store$.select(getMode);
+
   constructor(
     private readonly store: Store,
-    private route: ActivatedRoute
```
Comment (sislex) [inline projects/blink/src/lib/containers/blink-container/blink-container.component.ts:27]: add coma

File: projects/blink/src/lib/containers/blink-container/blink-container.component.ts (hunk 8 above)
Comment (sislex) [reply projects/blink/src/lib/containers/blink-container/blink-container.component.ts:34]: add coma

Comment (sislex) [reply projects/blink/src/lib/containers/blink-container/blink-container.component.ts:34]: private store$: Store<BlinkModeState> is better

File: projects/blink/src/lib/containers/blink-three-container/blink-three-container.component.html [hunk 22]
```diff
@@ -0,0 +1,6 @@
+<page-layout>
+  <blink-nav-panel-container  toplBar></blink-nav-panel-container>
```
Comment (sislex) [inline projects/blink/src/lib/containers/blink-three-container/blink-three-container.component.html:2]: two spaces

File: projects/blink/src/lib/containers/blink-two-container/blink-two-container.component.html [hunk 23]
```diff
@@ -0,0 +1,6 @@
+<page-layout>
+  <blink-nav-panel-container  toplBar></blink-nav-panel-container>
```
Comment (sislex) [inline projects/blink/src/lib/containers/blink-two-container/blink-two-container.component.html:2]: two spaces

**Pull request merged**